oauthenticator/openshift.py

```diff
@@ -7,18 +7,23 @@
 
 import json
 import os
-import requests
+import warnings
 
-from tornado.auth import OAuth2Mixin
+import requests
+from jupyterhub.auth import LocalAuthenticator
 from tornado import web
-
+from tornado.auth import OAuth2Mixin
+from tornado.httpclient import AsyncHTTPClient, HTTPClient, HTTPRequest
 from tornado.httputil import url_concat
-from tornado.httpclient import HTTPRequest, AsyncHTTPClient, HTTPClient
-from traitlets import Bool, Unicode, default
 
-from jupyterhub.auth import LocalAuthenticator
+from oauthenticator.oauth2 import OAuthenticator
+from traitlets import Bool, Set, Unicode, default
 
-from .oauth2 import OAuthenticator
+try:
+    import pycurl
+    AsyncHTTPClient.configure("tornado.curl_httpclient.CurlAsyncHTTPClient")
```

Comment:
This is already done in JupyterHub:
https://github.com/jupyterhub/jupyterhub/blob/d289cd1e025dfc35eec1d42df55cc2d274453b48/jupyterhub/app.py#L2321-L2330
and it should apply globally.

See the discussion in https://github.com/jupyterhub/oauthenticator/issues/385

I might be worth making this clearer in the docs though.

Reply:
@manics no worries, I can remove this block. Got caught up in testing for our hotfix and didn't test w/ `pycurl` installed w/o it.

Reply:
@manics I'll remove this, and yes a docs update would be great. I was misled a bit by the TODO inline comment so we may want to strip those as well.

Reply:
Could you create a new issue on this repo so it doesn't get forgotten :smile: ?